salt/metalk8s/kubeadm/init/mark-control-plane/configured.sls.in

```diff
@@ -3,23 +3,47 @@
 {% set kubeconfig = "/etc/kubernetes/admin.conf" %}
 {% set context = "kubernetes-admin@kubernetes" %}
 
-Apply control-plane role label:
+Apply control-plane master role label:
   kubernetes.node_label_present:
     - name: "node-role.kubernetes.io/master"
     - kubeconfig: {{ kubeconfig }}
     - context: {{ context }}
     - node: {{ hostname }}
     - value: ""
 
-Apply control-plane taints:
+Apply control-plane etcd role label:
+  kubernetes.node_label_present:
+    - name: "node-role.kubernetes.io/etcd"
+    - kubeconfig: {{ kubeconfig }}
+    - context: {{ context }}
+    - node: {{ hostname }}
+    - value: ""
+
+Apply control-plane bootstrap role label:
+  kubernetes.node_label_present:
+    - name: "node-role.kubernetes.io/bootstrap"
+    - kubeconfig: {{ kubeconfig }}
+    - context: {{ context }}
+    - node: {{ hostname }}
+    - value: ""
+
+Apply control-plane bootstrap taint:
   kubernetes.node_taints_present:
     - name: {{ hostname }}
     - kubeconfig: {{ kubeconfig }}
     - context: {{ context }}
     - taints:
-        - key: "node-role.kubernetes.io/master"
+        - key: "node-role.kubernetes.io/bootstrap"
           effect: "NoSchedule"
 
+Apply node version label:
+  kubernetes.node_label_present:
+    - kubeconfig: {{ kubeconfig }}
+    - context: {{ context }}
+    - name: "metalk8s.scality.com/version"
+    - node: {{ hostname }}
+    - value: "@@VERSION"
```

Comment:
For me we should have it somewhere else maybe in an sls a variable set in jinja so we can import it in any state and use it directly. 
Also this should also be a label for "workload nodes", no ?
Should maybe be in a dedicated state

Reply:
> For me we should have it somewhere else maybe in an sls a variable set in jinja so we can import it in any state and use it directly.

Hmh, maybe.

> Also this should also be a label for "workload nodes", no ?

No, bootstrap node is still 'special', not meant as a general-purpose node. If it needs to be a workload node, then the taint should be removed, and the role label applied.

> Should maybe be in a dedicated state

`mark-control-plane` needs to be reworked anyway :)

Reply:
> No, bootstrap node is still 'special', not meant as a general-purpose node. If it needs to be a workload node, then the taint should be removed, and the role label applied.

But that mean for a workload node we don't know the current metalk8s version installed ?

Reply:
How so? The version label is a prerequisite when the Node object is created. Only for the bootstrap node we need to apply the label, since for that one we still rely on self-registration.

Reply:
OHhhh yes it's true